### PR TITLE
[Documentation] Resource Layer and State Machine update

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | yes
| New docs?    | no
| BC breaks?      | no
| Related tickets | partially #5275 
| License         | MIT

+ ``Symfony2`` occurences -> ``Symfony``

### CODE REVIEW

File: docs/book/resource_layer.rst [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        $repository = $this->container->get('sylius.repository.product');
+
+        $products = $repository->createPaginator();
+        $products->setMaxPerPage(3);
+        $products->setCurrentPage($request->query->get('page', 1));
+
+        // Now you can return products to template and iterate over it to get products from current page.
+    }
+
+Paginator can be created for a specific criteria and with desired sorting.
```
Comment (dantleech) [inline docs/book/resource_layer.rst]: Should this and the previous two sentances before code blocks be terminated with `:` ?

File: docs/book/resource_layer.rst [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+Additionally, all these actions are very flexible and allow you to use different templates, forms, repository methods per route.
+The bundle is very powerful and allows you to register your own resources as well. To give you some idea of what is possible, here are some examples!
+
+Displaying a resource with a custom template and repository methods:
+
+.. code-block:: yaml
+
+    # routing.yml
+
```
Comment (dantleech) [inline docs/book/resource_layer.rst]: The symfony docs would not have a line break here: http://symfony.com/doc/current/book/controller.html#mapping-a-url-to-a-controller

File: docs/book/state_machine.rst [hunk 3]
```diff
@@ -15,15 +15,56 @@ States
 
 States of a state machine are defined as constants on the model of an entity that the state machine is controlling.
 
+How to configure states? Let's see on the example of our **Checkout** state machine.
```
Comment (dantleech) [inline docs/book/state_machine.rst]: Is that "our" like Sylius's or "our" like the users? I might say `Let's checkout the example from the **Checkout** ..` but unsure of the context.

File: docs/book/state_machine.rst [hunk 4]
```diff
@@ -15,15 +15,56 @@ States
...
+
+.. code-block:: yaml
+
+   winzou_state_machine:
```
Comment (dantleech) [inline docs/book/state_machine.rst]: Add path to this file?

File: docs/bundles/SyliusGridBundle/index.rst [hunk 5]
```diff
@@ -1,7 +1,7 @@
 SyliusGridBundle
 ================
 
-Displaying a grid with sorting and filtering options is a common task for many web applications. This bundle integrates the Sylius Grid component with Symfony2 framework and allows you to display grids really easily.
+Displaying a grid with sorting and filtering options is a common task for many web applications. This bundle integrates the Sylius Grid component with Symfony framework and allows you to display grids really easily.
```
Comment (dantleech) [inline docs/bundles/SyliusGridBundle/index.rst]: `... with the Symfony framework ...`

File: docs/bundles/SyliusRbacBundle/index.rst [hunk 6]
```diff
@@ -1,7 +1,7 @@
 SyliusRbacBundle
 ================
 
-Role-Based-Access-Control for Symfony2 applications, integrated with Doctrine ORM. Managable via UI.
+Role-Based-Access-Control for Symfony applications, integrated with Doctrine ORM. Managable via UI.
```
Comment (dantleech) [inline docs/bundles/SyliusRbacBundle/index.rst]: Ah, I always wondered what RBAC meant :) Maybe `Role Based Access Control (RBAC)` ? Don't *think* it is correct to hypenate here

Comment (TheMadeleine) [inline docs/bundles/SyliusRbacBundle/index.rst]: Thanks @dantleech for these remarks! :D although I'm not sure if I should change it in this PR. :D

**Comment by michalmarcinkowski** [comment]
👍  We should also add `$repository->add/remove` usages to Resource layer.

**Pull request merged**